lib/resource/resource.ex

```diff
@@ -3,30 +3,56 @@ defmodule AshPaperTrail.Resource do
   Documentation for `AshPaperTrail.Resource`.
   """
 
+  @belongs_to_actor %Spark.Dsl.Entity{
+    name: :belongs_to_actor,
+    describe: """
+    Creates a belongs_to relationship to version resource. When creating a new version, if the actor is set and matches the 
+    resource type, the version will be related to the actor. If your actors are polymorphic or varying types, declare a 
+    belongs_to_actor for each type.
+
+    The relationship created will always be `allow_nil? true` and foreign key constraints (if any) will unassociate 
+    if the actor record is deleted. If you need more complex relationship, set `define_attribute? false` and add the
+    relationship via a mixin.
+    """,
+    examples: [
+      "belongs_to_actor :user, MyApp.Users.User, api: MyApp.Users"
+    ],
+    no_depend_modules: [:destination],
+    target: AshPaperTrail.Resource.BelongsToActor,
+    args: [:name, :destination],
+    schema: AshPaperTrail.Resource.BelongsToActor.schema(),
+    transform: {AshPaperTrail.Resource.BelongsToActor, :transform, []}
+  }
```

Comment:
@zachdaniel thanks for the earlier feedback. Here's where I'm at now. 

Here's what I have in the test.
```elixir
paper_trail do 
  belongs_to_actor :user, MyApp.Accounts.User, api: MyApp.Accounts
  belongs_to_actor :news_feed, MyApp.Accounts.NewsFeed, api: MyApp.Accounts
end
```
I do want want to be able to declare this multiple times. 

here's the error that I'm currently getting:
```elixir
== Compilation error in file test/support/articles/article.ex ==
** (UndefinedFunctionError) function Spark.Dsl.Entity.fetch/2 is undefined (Spark.Dsl.Entity does not implement the Access behaviour. If you are using get_in/put_in/update_in, you can specify the field to be accessed using Access.key!/1)
    (spark 1.1.25) Spark.Dsl.Entity.fetch(%Spark.Dsl.Entity{name: :belongs_to_actor, target: AshPaperTrail.Resource.BelongsToActor, transform: {AshPaperTrail.Resource.BelongsToActor, :transform, []}, recursive_as: nil, examples: ["belongs_to_actor :user, MyApp.Users.User, api: MyApp.Users"], entities: [], singleton_entity_keys: [], deprecations: [], describe: "Creates a belongs_to relationship to version resource. When creating a new version, if the actor is set and matches the \nresource type, the version will be related to the actor. If your actors are polymorphic or varying types, declare a \nbelongs_to_actor for each type.\n\nThe relationship created will always be `allow_nil? true` and foreign key constraints (if any) will unassociate \nif the actor record is deleted. If you need more complex relationship, set `define_attribute? false` and add the\nrelationship via a mixin.\n", snippet: "", args: [:name, :destination], links: nil, hide: [], identifier: nil, modules: [], imports: [], no_depend_modules: [:destination], schema: [name: [type: :atom, doc: "The name of the relationship to use for the actor (e.g. :user)", required: true], destination: [type: {:spark, Ash.Resource}, doc: "The resource of the actor (e.g. MyApp.Users.User)"], api: [type: :atom, doc: "The API module to use when working with the related entity.\n"], define_attribute?: [type: :boolean, default: true, doc: "If set to `false` an attribute is not created on the resource for this relationship, and one must be manually added in `attributes`, invalidating many other options."]], auto_set_fields: [], docs: ""}, :hide)
```

My assumption is that I need to implement the transformer that you metioned in the previous comment but I'm not sure needs to be done.

Reply:
resolved. I forgot to remove @belongs_to_actor from the paper_trail schema and that was causing the error.